src/Config/FileConfig.php

```diff
@@ -34,16 +34,21 @@
      * @param bool $dev
      * @param EventDispatcherInterface $dispatcher
      * @param LoggerInterface $logger
+     * @param mixed[] $non_static_options
      */
     public function __construct(
         string $config_file,
         array $plugins = [],
         bool $dev = false,
         EventDispatcherInterface $dispatcher = null,
-        LoggerInterface $logger = null
+        LoggerInterface $logger = null,
+        array $non_static_options = []
```

Comment:
What is the use case for these non static options? Especially the docblock type `mixed` scares me.

Maybe we should remove the `FileConfig` and move the `ArrayConfig` from the bundle here instead. Or make the `FileConfig` class a wrapper around the `ArrayConfig` class.

Reply:
I had the issue I have one setting which is not statically determined. I can generate a json file and use this class or indeed make a wrapper class for just one setting or I could just add this argument to keep it simple

Reply:
seems to be only used in the test? There is an interface, just mock it?